daemon: Add "datapath" opt to --debug-verbose flag

brb opened the pull request:
This PR adds a support for the `"datapath"` option to `--debug-verbose`

Adding the `"datapath"` to `--debug-verbose` enables debug messages produced by the datapath.

Previously, it could have been enabled for new endpoints by invoking `cilium config Debug{,LB}=Enabled` on each node, and for existing - by `cilium config $EID Debug{,LB}=Enabled`.

Added `needs-backport/1.8`, as it will ease debugging of 1.8 user datapath issues.

```release-note
daemon: Add "datapath" opt to --debug-verbose flag to enable datapath debug messages
```

brb committed:
daemon: Add datapath option to --debug-verbose

Adding the "datapath" to --debug-verbose enables debug messages produced
by the datapath.

Previously, it could have been enabled for new endpoints by invoking
"cilium config Debug{,LB}=Enabled" on each node, and for existing -
by "cilium config $EID Debug{,LB}=Enabled".

Signed-off-by: Martynas Pumputis <m@lambda.lt>

pchaigno reviewed (approved):
`global.debugDatapath` made more sense to me, but I'm not sure it matters that much either way :shrug:

joestringer commented on install/kubernetes/cilium/charts/config/templates/configmap.yaml:
```diff
@@ -110,6 +110,10 @@ data:
 
+{{- if .Values.global.debug.datapath }}
+  debug-datapath: "{{ .Values.global.debug.datapath }}"
+{{- end }}
```
Shouldn't this be part of `config` rather than `global`?

brb replied:
I'm not very familiar with those groupings. What's a logical difference between `config` and `global`?

joestringer replied:
`config` is part of the settings specific to the helm chart that deploys the Cilium configmap; `global` applies settings to the entire set of helm charts we generate. Recent discussions with @cilium/helm particularly around the time of v1.8 release established that our use of Helm is a bit unusual to the detriment of operators, and one of the areas we can improve is in using more chart-local variables rather than global variables.

christarazi reviewed (commented):
Change looks good to me. I'd also like to +1 @borkmann's idea of reusing the same agent flags.

aanm reviewed (changes requested):
Please add a note in the upgrade guide as this is a breaking change. Previous environments running with `debug: true` will no longer see any debug message when running with `cilium monitor -v`

aanm commented on daemon/cmd/daemon_main.go:
```diff
@@ -1013,8 +1019,8 @@ func initEnv(cmd *cobra.Command) {
 
-	option.Config.Opts.SetBool(option.Debug, false)
-	option.Config.Opts.SetBool(option.DebugLB, false)
+	option.Config.Opts.SetBool(option.Debug, debugDatapath)
```
Also, perhaps it would be a good idea to add a note in the flag description of the `debug` flag that it no longer enables debug messages in the datapath and it should users should use `debug-verbose=datapath` instead.

joestringer replied:
^ Probably needs backport right?

aanm replied:
@joestringer this PR is already marked for backport, was that your concern?

aanm reviewed (changes requested):
Also, please add `--debug-verbose=datapath` in the `contrib/vagrant/start.sh` where we have `--debug`

aanm reviewed (changes requested):
I wasn't aware that https://github.com/cilium/cilium/commit/ee5e396a9b6c8244b083124c0f00bab3d7a96059 introduced this breaking change. Given that it did, yes we should describe it in the upgrade guide. If we start helping users where we ask them to run cilium with `debug: true` so that we can see debug messages in Cilium monitor we won't see debug messages and would waste more time until we could realize this was introduced by us.

christarazi reviewed (approved):
LGTM assuming Andre's comments are addressed

brb committed:
docs: Add note about enabling datapath dbg messages

Add a note to the upgrade guide how to enable datapath debug messages
which were disabled by [1] (released in v1.8.0).

[1]: https://github.com/cilium/cilium/commit/583e67c912e26f7e0bed2656d9ae7544a8d132af

Signed-off-by: Martynas Pumputis <m@lambda.lt>
